src/catchup/CatchupWork.cpp

```diff
@@ -221,12 +210,7 @@ CatchupWork::doWork()
     // Step 2: Compare local and remote states
     if (!hasAnyLedgersToCatchupTo())
     {
-        mApp.getCatchupManager().historyCaughtup();
-        asio::error_code ec = std::make_error_code(std::errc::invalid_argument);
-        mProgressHandler(ec, ProgressState::FINISHED,
-                         LedgerHeaderHistoryEntry{},
-                         mCatchupConfiguration.mode());
-        return State::WORK_SUCCESS;
+        return State::WORK_FAILURE;
```

Comment:
why not logging here instead of in `onFailureRaise` ? This is where we have full context (later you have to reverse engineer the failure)